src/plugins/template/src/template_plugin.cpp

```diff
@@ -2,85 +2,65 @@
     // TODO: fill with actual device name, backend engine
-    _pluginName = "TEMPLATE";
+    set_device_name("TEMPLATE");
 
     // create ngraph backend which performs inference using ngraph reference implementations
     _backend = ngraph::runtime::Backend::create();
 
     // create default stream executor with a given name
-    _waitExecutor = executorManager()->getIdleCPUStreamsExecutor({"TemplateWaitExecutor"});
+    _waitExecutor = get_executor_manager()->getIdleCPUStreamsExecutor({"TemplateWaitExecutor"});
 }
 // ! [plugin:ctor]
 
 // ! [plugin:dtor]
 Plugin::~Plugin() {
     // Plugin should remove executors from executor cache to avoid threads number growth in the whole application
-    executorManager()->clear("TemplateStreamsExecutor");
-    executorManager()->clear("TemplateWaitExecutor");
+    get_executor_manager()->clear("TemplateStreamsExecutor");
+    get_executor_manager()->clear("TemplateWaitExecutor");
     // NOTE: Uncomment this if Inference Engine Executor cache is used to create callback executor
     // executorManager()->clear("TemplateCallbackExecutor");
 }
 // ! [plugin:dtor]
 
-// ! [plugin:transform_network]
+ov::RemoteContext TemplatePlugin::Plugin::create_context(const ov::AnyMap& remote_properties) const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+ov::RemoteContext TemplatePlugin::Plugin::get_default_context(const ov::AnyMap& remote_properties) const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
 
-void TransformNetwork(std::shared_ptr<ngraph::Function>& function,
-                      const InferenceEngine::InputsDataMap& inputInfoMap,
-                      const InferenceEngine::OutputsDataMap& outputsInfoMap) {
+// ! [plugin:transform_network]
+void transform_model(const std::shared_ptr<ov::Model>& model) {
     // Perform common optimizations and device-specific transformations
-    ngraph::pass::Manager passManager;
-    // Example: register transformation to convert preprocessing information to graph nodes
-    passManager.register_pass<ngraph::pass::AddPreprocessing>(inputInfoMap);
-    // TODO: add post-processing based on outputsInfoMap
+    ov::pass::Manager passManager;
     // Example: register CommonOptimizations transformation from transformations library
     passManager.register_pass<ov::pass::CommonOptimizations>();
-    // G-API supports only FP32 networks for pre-processing
-    bool needF16toF32 = false;
-    for (const auto& param : function->get_parameters()) {
-        if (param->get_element_type() == ngraph::element::f16 &&
-            inputInfoMap.at(param->get_friendly_name())->getTensorDesc().getPrecision() !=
-                InferenceEngine::Precision::FP16) {
-            needF16toF32 = true;
-            break;
-        }
-    }
-    if (needF16toF32) {
-        passManager.register_pass<ov::pass::ConvertPrecision>(
-            precisions_array{{ngraph::element::f16, ngraph::element::f32}});
-    }
+    passManager.get_pass_config()->disable<ov::pass::ConvertReduceSumToPooling>();
```

Comment:
is it just an example? Let's insert explaining comments why we are doing it

Reply:
No it is a bug in transformations